packages/splits-sdk/src/client/splitV2.ts

```diff
@@ -0,0 +1,1112 @@
+  TransactionType,
+  getSplitV2FactoriesStartBlock,
+  getSplitV2FactoryAddress,
+} from '../constants'
+import { splitV2ABI } from '../constants/abi/splitV2'
+import { splitV2FactoryABI } from '../constants/abi/splitV2Factory'
+import {
+  SaltRequired,
+  TransactionFailedError,
+  UnsupportedChainIdError,
+} from '../errors'
+import {
+  CallData,
+  CreateSplitV2Config,
+  DistributeSplitConfig,
+  SetPausedConfig,
+  Split,
+  SplitV2ExecCallsConfig,
+  SplitV2Type,
+  SplitsClientConfig,
+  TransactionConfig,
+  TransactionFormat,
+  TransferOwnershipConfig,
+  UpdateSplitV2Config,
+} from '../types'
+import {
+  fromBigIntToPercent,
+  getNumberFromPercent,
+  getSplitType,
+  getValidatedSplitV2Config,
+  validateAddress,
+} from '../utils'
+import {
+  BaseClientMixin,
+  BaseGasEstimatesMixin,
+  BaseTransactions,
+} from './base'
+import { applyMixins } from './mixin'
+
+type SplitFactoryABI = typeof splitV2FactoryABI
+type SplitV2ABI = typeof splitV2ABI
+
+const VALID_ERC1271_SIG = '0x1626ba7e'
+
+// TODO:add validation to execute contract function
+class SplitV2Transactions extends BaseTransactions {
+  constructor({
+    transactionType,
+    chainId,
+    publicClient,
+    ensPublicClient,
+    walletClient,
+    apiConfig,
+    includeEnsNames = false,
+  }: SplitsClientConfig & TransactionConfig) {
+    super({
+      transactionType,
+      chainId,
+      publicClient,
+      ensPublicClient,
+      walletClient,
+      apiConfig,
+      includeEnsNames,
+    })
+  }
+
+  protected async _createSplit({
+    recipients,
+    distributorFeePercent,
+    totalAllocationPercent,
+    splitType = SplitV2Type.Pull,
+    controllerAddress = zeroAddress,
+    creatorAddress = zeroAddress,
+    salt,
+    transactionOverrides = {},
+  }: CreateSplitV2Config): Promise<TransactionFormat> {
+    const {
+      recipientAddresses,
+      recipientAllocations,
+      distributionIncentive,
+      totalAllocation,
+    } = getValidatedSplitV2Config(
+      recipients,
+      distributorFeePercent,
+      totalAllocationPercent,
+    )
+
+    recipientAddresses.map((recipient) => validateAddress(recipient))
+    validateAddress(controllerAddress)
+    validateAddress(creatorAddress)
+
+    this._requirePublicClient()
+    if (this._shouldRequireWalletClient) this._requireWalletClient()
+
+    const functionName = salt ? 'createSplitDeterministic' : 'createSplit'
+    const functionArgs = [
+      {
+        recipients: recipientAddresses,
+        allocations: recipientAllocations,
+        totalAllocation,
+        distributionIncentive,
+      },
+      controllerAddress,
+      creatorAddress,
+    ]
+    if (salt) functionArgs.push(salt)
+
+    return this._executeContractFunction({
+      contractAddress: getSplitV2FactoryAddress(this._chainId, splitType),
+      contractAbi: splitV2FactoryABI,
+      functionName,
+      functionArgs,
+      transactionOverrides,
+    })
+  }
+
+  protected async _transferOwnership({
```

Comment:
nit: maybe a `_requireOwner` function for all these owner only tx's? just to give a more friendly error message.